File: PythonProjects/PythonCoreLib/Redis/RedisClient.py
import redis

class RedisClient(object):
    def __init__(self, host, port):
        self.ConnectionPool = redis.ConnectionPool(host = host, port = port)
        self.client = redis.Redis(connection_pool = self.ConnectionPool)

    def set(self, key, value):
        self.client.set(key, value)

    def get(self, key):
        return self.client.get(key)

    def hset(self, name, k, v):
        self.client.hset(name, k, v)

    def hget(self, name, k):
        return self.client.hget(name, k)

    def lpush(self, name, value):
        self.client.lpush(name, value)

    def lpop(self, name, start=-1, end=-1):
        data = self.client.lpop(name)
        if 0 <= start <= end:
            return data[start:end]
        return data

File: Wwbweibo.CrackDetect.Libs/Redis/RedisClient.cs
using StackExchange.Redis;
using System.Linq;
using Wwbweibo.CrackDetect.Libs.Tools.String;

namespace Wwbweibo.CrackDetect.Libs.Redis
{
    public class RedisClient : IRedisClient
    {
        private static ConnectionMultiplexer connection;
        private static IDatabase database;
        private static IServer redisServer;

        public RedisClient(string server, string port)
        {
            connection = ConnectionMultiplexer.Connect($"{server}:{port}");
            database = connection.GetDatabase();
            redisServer = connection.GetServer(server, port.ToInt());
        }

        public void Set(string key, string value)
        {
            database.StringSet(key, value);
        }

        public string Get(string key)
        {
            return database.StringGet(key);
        }

        public string[] LPop(string key, int start = -1, int end = -1)
        {
            if (start >= 0 && end >= start)
                return database.ListRange(key, start, end).ConvertValuesToStrings();
            return database.ListRange(key).ConvertValuesToStrings();
        }

        public void LPush(string key, string[] values)
        {

            database.ListRightPush(key, values.ConvertStringsToRedisValues());
        }

        public void HSet(string key, string name, string value)
        {
            database.HashSet(key, name, value);
        }

        public string HGet(string key, string name)
        {
            return database.HashGet(key, name);
        }
    }

    public static class RedisValueExtensions
    {
        public static string[] ConvertValuesToStrings(this RedisValue[] values)
        {
            return values.Select(p => p.ToString()).ToArray();
        }

        public static RedisValue[] ConvertStringsToRedisValues(this string[] values)
        {
            return values.Select(p => (RedisValue)p).ToArray();
        }
    }
}
